include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+// Function _callServer()
+// ----------------------------------------------------------------------------
+inline bool _callServer(VersionCheck me, bool version_file_exists)
+{
+    // system call
+    // http response is stored in a file '.config/seqan/{app_name}_version'
+    if (system(me._command.c_str()))
+    {
+        std::string version_file = me._path + "/" + me._name + "_version.txt";
+        if (!version_file_exists)
+        {
+            std::ofstream version_file_out (version_file.c_str());
+            if (version_file_out.is_open())
+            {
+                version_file_out << "UNABLE TO CALL HOME.\n";
+                version_file_out.close();
+            }
+        }
+        else // just set last modified date to current and do not override possible version information
+        {
+#if defined(PLATFORM_WINDOWS)
+            FILETIME curr;
+            GetSystemTimeAsFileTime(&curr);
+            HANDLE hFile;
+            hFile = CreateFile(version_file.c_str(), GENERIC_WRITE, FILE_SHARE_READ,  NULL,  OPEN_EXISTING,  FILE_ATTRIBUTE_NORMAL, NULL);
+            SetFileTime(hFile, NULL, NULL, &curr);
+            CloseHandle(hFile);
+#else
+            time_t curr;
+            time(&curr); // get current time
+            struct utimbuf puttime;
+            puttime.modtime = curr;
+            puttime.actime = curr;
+            utime(version_file.c_str(), &puttime);
+#endif
+        }
+        return false;
+    }
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function checkForNewerVersion()
+// ----------------------------------------------------------------------------
+inline bool checkForNewerVersion(VersionCheck & me)
+{
+    if (!_checkWritability(me._path))
+    {
+#if defined(PLATFORM_WINDOWS)
+        TCHAR tmp_path [MAX_PATH];
+        if (GetTempPath (MAX_PATH, tmp_path) != 0)
+            me._path = tmp_path;
+        else //GetTempPath() returns 0 on failure
+            return false;
+# else // unix
+        me._path = "/tmp";
+#endif
+        me._updateCommand();
+    }
+
+    std::string version_file = me._path + "/" + me._name + "_version.txt";
+    double min_time_diff = 86400;                           // one day = 86400 seonds
+    double file_time_diff = _getFileTimeDiff(version_file); // time difference: last modified date until now in seconds
+    bool version_file_exists = (file_time_diff > -1) ? true : false;
+
+    if (file_time_diff < min_time_diff && version_file_exists)
+        return false; // only check for newer version once a day
+    
+    if (version_file_exists) // file exists. TODO:: ask if there should be a time limit here too
+    {
+        seqan::String<int> new_ver;
+        if (_readVersionNumbers(new_ver, version_file))
+        {
+            seqan::String<int> old_ver;
+            _getNumbersFromString(old_ver, me._version);
+
+            if (_isSmaller(old_ver, new_ver))
```

Comment:
You could use SeqAn's  Comparator instead of writing your own method.
This stores the result and you have to parse only once. See here the documentation: http://docs.seqan.de/seqan/develop/?p=Lexical
You can then ask the comparator object ```isLess() ``` or ```isGreater()``` for the result.